src/bootstrap/native.rs

```diff
@@ -465,6 +470,30 @@ impl Step for Llvm {
 
         cfg.build();
 
+        // When building LLVM with LLVM_LINK_LLVM_DYLIB for macOS, an unversioned
+        // libLLVM.dylib will be built. However, llvm-config will still look
+        // for a versioned path like libLLVM-14.dylib. Manually create a symbolic
+        // link to make llvm-config happy.
+        #[cfg(unix)]
```

Comment:
Please don't add cfgs like this - looking at the target specifications is enough. We want this to work when cross compiling.

Reply:
The reason here is that it relies on `std::os::unix::symlink`, which is only available on Unix platforms

Reply:
The windows equivalent is https://doc.rust-lang.org/stable/std/os/windows/fs/fn.symlink_file.html.

Reply:
I have added a branch for cross compiling on Windows and will panic if the build script is not running on either Windows or Unix.